packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -231,12 +231,15 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
               // check the chunk being imported
               const url = code.slice(start, end)
               const deps: Set<string> = new Set()
+              const analyzed: Set<string> = new Set<string>()
 
               if (url[0] === `"` && url[url.length - 1] === `"`) {
                 const ownerFilename = chunk.fileName
                 // literal import - trace direct imports and add to deps
                 const addDeps = (filename: string) => {
                   if (filename === ownerFilename) return
+                  if (analyzed.has(filename)) return
+                  analyzed.add(filename)
```

Comment:
Thanks for reworking the PR and adding the tests.

About the name of the test, we may want to test other things and not only this infinite loop. Probably better to call it multiple-entrypoints or something that indicates that it is intended to stress test the mpa use case.

We should move analyzed inside the branch, as it is not used out of it:
```suggestion
              if (url[0] === `"` && url[url.length - 1] === `"`) {
                const ownerFilename = chunk.fileName
                // literal import - trace direct imports and add to deps
                const analyzed: Set<string> = new Set<string>()
                const addDeps = (filename: string) => {
                  if (filename === ownerFilename) return
                  if (analyzed.has(filename)) return
                  analyzed.add(filename)
```

Reply:
Jep, good points. I changed the playground name as suggested and moved analyzed inside the if.